test/cat.jl

```diff
@@ -224,62 +224,24 @@ end
 
 @testset "vcat" begin
     missing_df = DataFrame()
-    df = DataFrame(Matrix{Int}(undef, 4, 3))
-
-    # Assignment of rows
-    # TODO: re-enable when we fix setindex! to handle DataFrameRow on RHS
-    # df[1, :] = df[1, :]
-    df[1, :] = df[1:1, :]
-    df[1:2, :] = df[1:2, :]
-    df[[true,false,false,true], :] = df[2:3, :]
-
-    # Scalar broadcasting assignment of rows
-    df[1, :] = 1
-    df[1:2, :] = 1
-    df[[true,false,false,true], :] = 3
-
-    # Vector broadcasting assignment of rows
-    df[1:2, :] = [2,3]
-    df[[true,false,false,true], :] = [2,3]
-
-    # Assignment of columns
-    df[1] = zeros(4)
-    df[:, 2] = ones(4)
-
-    # Broadcasting assignment of columns
-    df[:, 1] = 1
-    df[1] = 3
-    df[:x3] = 2
-
-    # assignment of subtables
-    # TODO: re-enable when we fix setindex! to handle DataFrameRow on RHS
-    # df[1, 1:2] = df[2, 2:3]
-    df[1, 1:2] = df[2:2, 2:3]
-    df[1:2, 1:2] = df[2:3, 2:3]
-    df[[true,false,false,true], 2:3] = df[1:2,1:2]
-
-    # scalar broadcasting assignment of subtables
-    df[1, 1:2] = 3
-    df[1:2, 1:2] = 3
-    df[[true,false,false,true], 2:3] = 3
-
-    # vector broadcasting assignment of subtables
-    df[1:2, 1:2] = [3,2]
-    df[[true,false,false,true], 2:3] = [2,3]
+    df = DataFrame([3.0  2.0  2.0
+                    2.0  2.0  2.0
+                    3.0  1.0  2.0
+                    3.0  3.0  3.0])
 
     @test vcat(missing_df) == DataFrame()
     @test vcat(missing_df, missing_df) == DataFrame()
     @test vcat(missing_df) == DataFrame()
     @test vcat(missing_df, missing_df) == DataFrame()
     @test vcat(missing_df, df) == df
     @test vcat(df, missing_df) == df
-    @test eltypes(vcat(df, df)) == Type[Float64, Float64, Int]
+    @test eltypes(vcat(df, df)) == Type[Float64, Float64, Float64]
```

Comment:
Maybe keep the last column as `Int` just in case the code didn't handle the heterogeneity correctly.

Reply:
OK. Fixed.